### PR TITLE
Support 2304: KernelSU & Remove amazon

### PR DESCRIPTION
**Pull request by s1204IT**
### Checklist

- [x] Referenced all related issues in the PR body
  - https://github.com/tiann/KernelSU/pull/475
  - #526
- [x] Have tested the modifications

---

Awaiting response.

### CODE REVIEW

**Comment by YT-Advanced** [comment]
@s1204IT #526 He use Magisk instead of KernelSU. So did you test your new script with Magisk ? Also the problem is EROFS not KernelSU.

**Comment by s1204IT** [comment]
> @s1204IT #526 He use Magisk instead of KernelSU. So did you test your new script with Magisk ? Also the problem is EROFS not KernelSU.

I know that.
Since it is not possible to fix and write to the system image file, we verified KernelSU as an alternative measure at this time.

This is a "**workaround**".
There is nothing better than Magisk at working root solution.

**Comment by s1204IT** [comment]
<details><summary><b>Comment</b></summary>
I'm sorry if my English sounded overbearing.

Please forgive me.
</details>

**Comment by YT-Advanced** [comment]
> I know that.
Since it is not possible to fix and write to the system image file, we verified KernelSU as an alternative measure at this time.
This is a "workaround".
There is nothing better than Magisk at working root solution.

@s1204IT But good new is Microsoft have reverted v2304. We cannot find it in Fast Channel now (also with Insider User)

**Comment by s1204IT** [comment]
> > I know that.
> > Since it is not possible to fix and write to the system image file, we verified KernelSU as an alternative measure at this time.
> > This is a "workaround".
> > There is nothing better than Magisk at working root solution.
> 
> @s1204IT But good new is Microsoft have reverted v2304. We cannot find it in Fast Channel now (also with Insider User)

Really?
I'll check later

**Comment by s1204IT** [comment]
> > > I know that.
> > > Since it is not possible to fix and write to the system image file, we verified KernelSU as an alternative measure at this time.
> > > This is a "workaround".
> > > There is nothing better than Magisk at working root solution.
> > 
> > 
> > @s1204IT But good new is Microsoft have reverted v2304. We cannot find it in Fast Channel now (also with Insider User)
> 
> Really? I'll check later


🤔 
I found it.

**Comment by s1204IT** [comment]
@YT-Advanced I'll test it in your repository.

**Comment by YT-Advanced** [comment]
> @YT-Advanced I'll test it in your repository.

Yes, I saw your second run in WSA have the output is v2303 instead of v2304. So I think maybe Microsoft revert it but nope

**Comment by MustardChef** [comment]
For a moment I thought that this was a full fix for the Erofs issue

**Comment by YT-Advanced** [comment]
> For a moment I thought that this was a full fix for the Erofs issue

This is just a "workaround". Because KernelSU don't need to modify the vhdx file, just paste kernel file to WSA Kernel

**Comment by s1204IT** [comment]
> > @YT-Advanced I'll test it in your repository.
> 
> Yes, I saw your second run in WSA have the output is v2303 instead of v2304. So I think maybe Microsoft revert it but nope

Confirmation, how did you get the user code?
Even if  capturing, it does not appear as a "user" value as before.

File: scripts/build.sh [hunk 1]
```diff
@@ -727,6 +729,7 @@ if [ "$REMOVE_AMAZON" ]; then
     echo "Remove Amazon Appstore"
     find "${PRODUCT_MNT:?}"/{etc/permissions,etc/sysconfig,framework,priv-app} 2>/dev/null | grep -e amazon -e venezia | sudo xargs rm -rf
     find "${SYSTEM_EXT_MNT:?}"/{etc/*permissions,framework,priv-app} 2>/dev/null | grep -e amazon -e venezia | sudo xargs rm -rf
+    rm -rf "$WORK_DIR/wsa/$ARCH/apex"
```
Comment (YT-Advanced) [inline scripts/build.sh:732]: Why you don't use it with sudo ?

Comment (Howard20181) [inline scripts/build.sh:732]: Do not remove any [apex](https://source.android.com/docs/core/ota/apex), which will cause system exceptions.

**Comment by s1204IT** [comment]
@Howard20181 
This works as expected.

**Review by Howard20181 — APPROVED** [review]

**Pull request merged**